compiler/rustc_lint/messages.ftl

```diff
@@ -440,17 +440,19 @@ lint_non_local_definitions_cargo_update = the {$macro_kind} `{$macro_name}` may
 
 lint_non_local_definitions_deprecation = this lint may become deny-by-default in the edition 2024 and higher, see the tracking issue <https://github.com/rust-lang/rust/issues/120363>
 
-lint_non_local_definitions_impl = non-local `impl` definition, they should be avoided as they go against expectation
+lint_non_local_definitions_impl = non-local `impl` definition, `impl` blocks should be written at the same level as their item
     .help =
-        move this `impl` block outside the of the current {$body_kind_descr} {$depth ->
+        move this `impl` block and outside the of the current {$body_kind_descr} {$depth ->
             [one] `{$body_name}`
            *[other] `{$body_name}` and up {$depth} bodies
         }
-    .non_local = an `impl` definition is non-local if it is nested inside an item and may impact type checking outside of that item. This can be the case if neither the trait or the self type are at the same nesting level as the `impl`
-    .exception = one exception to the rule are anon-const (`const _: () = {"{"} ... {"}"}`) at top-level module and anon-const at the same nesting as the trait or type
+    .without_trait = methods and assoc const are still usable outside the current expression, only `impl Local` and `impl dyn Local` are local and only if the `Local` type is at the same nesting as the `impl` block
```

Comment:
I think this might be clearer? slightly reduces [semantic satiation](https://en.wikipedia.org/wiki/Semantic_satiation) at least.
```suggestion
    .without_trait = methods and assoc const are still usable outside the current expression, `impl Local` and `impl dyn Local` are only private if the `Local` type is nested in the same item as the `impl`
```

Reply:
unfortunately this also changes the meaning significantly

what this note is trying to say is that only `impl Local` and `impl dyn Local` can ever be private, and of those two, to be private, `Local` must be at the same nesting as the `impl` def

Reply:
hmm so closer to this?
```suggestion
    .without_trait = methods and assoc const are still usable outside the current expression, only `impl Local` and `impl dyn Local` can be private if the type is nested in the same item as the `impl`
```

Reply:
( it's not required to find a different wording I'm just trying to figure things out. )

Reply:
idk, feels a bit weird, at least when I try to read it with my non-native english eyes

maybe this?
```suggestion
    .without_trait = methods and associated constants are still usable outside the current expression, only `impl Local` and `impl dyn Local` can ever be private, and only if the type is nested in the same item as the `impl`
```

Reply:
Ooh, that seems good.

Reply:
Done.